httpcore5/src/main/java/org/apache/hc/core5/http/protocol/RequestTE.java

```diff
@@ -0,0 +1,184 @@
+ * values (denoted by {@code q=}) separated by semicolons.
+ * <p>
+ * In case of HTTP/2, this validation is skipped, and another layer of logic handles the specifics of HTTP/2 compliance.
+ *
+ * @since 5.4
+ */
+@Contract(threading = ThreadingBehavior.IMMUTABLE)
+public class RequestTE implements HttpRequestInterceptor {
+
+    /**
+     * Singleton instance of the {@code RequestTE} interceptor.
+     */
+    public static final HttpRequestInterceptor INSTANCE = new RequestTE();
+
+    /**
+     * Delimiter used to parse the {@code TE} header, recognizing both commas (',') and semicolons (';') as delimiters.
+     */
+    public static final Tokenizer.Delimiter DELIMITER = Tokenizer.delimiters(',', ';');
+
+    /**
+     * Default constructor.
+     */
+    public RequestTE() {
+        super();
+    }
+
+    /**
+     * Processes the {@code TE} header of the given HTTP request and ensures compliance with HTTP/1.1 requirements.
+     * <p>
+     * If the {@code TE} header is present, this method validates that:
+     * <ul>
+     * <li>The {@code TE} header does not include the {@code chunked} transfer coding, which is implicitly supported for HTTP/1.1.</li>
+     * <li>The {@code Connection} header includes the {@code TE} directive, as required by the protocol.</li>
+     * </ul>
+     *
+     * @param request the HTTP request containing the headers to validate
+     * @param entity  the entity associated with the request (may be {@code null})
+     * @param context the execution context for the request
+     * @throws HttpException if the {@code TE} header contains invalid values or the {@code Connection} header is missing
+     * @throws IOException   in case of an I/O error
+     */
+    @Override
+    public void process(final HttpRequest request, final EntityDetails entity, final HttpContext context)
+            throws HttpException, IOException {
+        Args.notNull(request, "HTTP request");
+
+        // Fetch the TE header
+        final Header teHeader = request.getFirstHeader(HttpHeaders.TE);
+
+        if (teHeader == null) {
+            return;  // No further validation needed
+        }
+
+        final String teValue = teHeader.getValue();
+        validateTEField(teValue);
+        validateConnectionHeaders(request);
+    }
+
+    /**
+     * Validates the {@link HttpHeaders#TE} header values for compliance with HTTP/1.1.
+     * <p>
+     * Specifically, this method ensures that:
+     * <ul>
+     * <li>The {@link HttpHeaders#TE} header does not contain the {@code chunked} transfer coding.</li>
+     * <li>The {@code trailers} directive is allowed and treated as valid.</li>
+     * </ul>
+     *
+     * @param teValue the value of the {@code TE} header
+     * @throws HttpException if the {@code TE} header contains invalid values
+     */
+    private void validateTEField(final String teValue) throws HttpException {
```

Comment:
@arturobernalg `TE` tokens can also be transmitted with multiple headers, like `Connection`. 

```
TE: this
TE: that, trailers
```

Can you change this method to use `MessageSupport#parseTokens`?

Reply:
done.

Reply:
> done.

@arturobernalg Your current approach results in an unnecessary creation of an array and a copy of the header value, which is wasteful. Please use `MessageSupport#parseTokens` that avoids all that.